packages/base/src/features/OpenUI5Support.ts

```diff
@@ -29,48 +29,73 @@ type OpenUI5CoreConfiguration = {
 	}
 };
 
+type ControlBehavior = {
+	getAnimationMode: () => string,
+}
+
+type Localization = {
+	getLanguage: () => string,
+	getLanguageTag: () => string,
+	getRTL: () => string,
+	getTimezone: () => string,
+}
+
 type LocaleData = {
 	getInstance: (locale: string) => Locale,
 }
 
-type CoreTheming = {
-    getThemeRoot: () => string,
+type Theming = {
+	getThemeRoot: () => string,
+	getTheme: () => string,
+	attachApplied: (callback: () => Promise<void>) => void,
+}
+
+type Formatting = {
+	getCalendarType: () => string,
+	getLegacyDateCalendarCustomizing: () => LegacyDateCalendarCustomizing,
+}
+
+type CalendarUtils = {
+	getWeekConfigurationValues: () => {
+		firstDayOfWeek: number | undefined,
+	},
 }
 
 type Locale = {
 	getFirstDayOfWeek: () => number,
 	_get: () => CLDRData,
 };
 
-type VersionUtil = (version: string) => {
-	compareTo: (version: string) => number,
-};
-
-const getCore = () => {
-	return window.sap?.ui?.getCore?.() as OpenUI5Core;
-};
-
 class OpenUI5Support {
+	static isModularCore() {
+		const version = window.sap.ui!.version as string;
+		const parts = version.split(".");
+		return parts && parts[1] && parseInt(parts[1]) >= 116;
+	}
+
 	static isLoaded() {
-		return !!getCore();
+		return typeof window.sap?.ui?.getCore === "function";
 	}
 
 	static init() {
-		const core = getCore();
-		if (!core) {
+		if (!OpenUI5Support.isLoaded()) {
 			return Promise.resolve();
 		}
 
 		return new Promise<void>(resolve => {
-			const deps: Array<string> = ["sap/ui/core/Popup", "sap/ui/core/LocaleData"];
-			const version: string = window.sap.ui.version || "";
-
-			window.sap.ui.require(["sap/base/util/Version"], (VersionUtil: VersionUtil) => {
-				if (VersionUtil(version).compareTo("1.116.0") >= 0) { // for versions since 1.116.0 and onward, use the Theming module
-					deps.push("sap/ui/core/Theming");
-				}
-
-				core.attachInit(() => {
+			window.sap.ui.require(["sap/ui/core/Core"], (Core: OpenUI5Core) => {
+				Core.attachInit(() => {
```

Comment:
We introduced a new Core.ready() function returning a Promise. You should use the ready instead of attachInit here. This will then work for both bootstraps.

Reply:
I couldn't get `Core.ready` to work with the old version (only with the 1.116 version) so I branched the logic.

Reply:
What exactly is not working? You could also pass a function (Core.ready(callback)) wich should behave exactly like attachInit...

Reply:
As I understood it, Vladi had issues on the _version_ dimension: 1.115 does not support `Core.ready()` yet.
In other words, `isModularCore()` in the code above checks for version 1.116, not for "is CoreX used".

Reply:
Indeed, I meant that `Core.ready` can only be used with the newest version, therefore I must keep the old code for 1.115 and earlier.